javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java

```diff
@@ -100,7 +100,7 @@ private static void optionalParametersJavadoc(String title, List<ProxyMethodPara
         for (ProxyMethodParameter parameter : parameters) {
             commentBlock.line(String.format(
                     "    <tr><td>%s</td><td>%s</td><td>%s</td><td>%s</td></tr>",
-                    parameter.getName(),
+                    parameter.getRequestParameterName(),
```

Comment:
nice catch!

Reply:
Do you know whether BODY parameter would show up here?
BODY would have `parameter.getRequestParameterName() == null`. So if it could be here, you need to have logic to fallback to `parameter.getName()`

Reply:
BODY parameter will not show up here because it is filtered out in the previous steps:
https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java#L39-L51